src/hangman/models/game.model.ts

```diff
@@ -54,12 +58,13 @@ export class Game extends AggregateRoot {
     this.playerId = data.playerId;
     this.wordToGuess = data.wordToGuess;
     this.maxGuesses = data.maxGuesses;
-    this.lettersGuessed = [];
+    // this.lettersGuessed = []; // Moved this initilization to the constructor
     this.dateCreated = new Date();
     this.dateModified = new Date();
 
     try {
-      await validateOrReject(this);
+      // Does this need to be async with await? Can this entire method be made sync?
```

Comment:
yes, `validateOrReject` returns a promise. I think you should set the aggregate with the data, then validate the state of the aggregate

Reply:
Yea you're right, I checked the docs for this package.